src/modules/main/components/NewestContainer.tsx

```diff
@@ -1,53 +1,93 @@
-import React, { useEffect } from 'react';
+import React, { useEffect, useRef } from 'react';
 import { useSelector, useDispatch } from 'react-redux';
-import { Container, Grid, Button, Typography } from '@material-ui/core';
+import {
+  Container,
+  Grid,
+  Button,
+  Typography,
+  CircularProgress,
+} from '@material-ui/core';
 import { RootStateType } from '../../../store/rootReducer';
 import { fetchNewestPosts, IMainState } from '../store/mainSlice';
-import PostPreviewCard from '../../../lib/components/PostPreview/PostPreviewCard';
 import { useStyles } from './styles/NewestContainer.style';
 import BorderBottom from '../../../lib/components/Border';
+import PostsList from '../../../lib/components/PostsList';
 
 const NewestContainer: React.FC = () => {
   const classes = useStyles();
+  const gridRef = useRef<HTMLDivElement>(null);
 
   const dispatch = useDispatch();
   const setNewest = () => dispatch(fetchNewestPosts());
 
-  const { newestPosts, meta } = useSelector<RootStateType, IMainState['newest']>(
-    (state) => {
-      return state.main.newest;
-    },
-  );
+  const {
+    newestPosts,
+    meta: { isLastPage, loading },
+  } = useSelector<RootStateType, IMainState['newest']>((state) => {
+    return state.main.newest;
+  });
 
   useEffect(() => {
     setNewest();
   }, []);
 
+  const renderLoading = () => {
+    return loading === 'pending' ? <CircularProgress /> : null;
+  };
```

Comment:
```suggestion
  const renderLoading = () =>loading === 'pending' ? <CircularProgress /> : null;
```
1-liners are better here :)

Reply:
fixed